File: EntreBicis_Backend/src/main/java/cat/copernic/mbotana/entrebicis_backend/entity/Reservation.java
package cat.copernic.mbotana.entrebicis_backend.entity;

import java.time.LocalDateTime;

import cat.copernic.mbotana.entrebicis_backend.entity.enums.ReservationState;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private ReservationState reservationState;

    @Column
    private LocalDateTime reservationTime;

    @Column
    private LocalDateTime returnTime;

    @ManyToOne
    @JoinColumn(name = "user_id", unique = true)
    private User user;

    @OneToOne
    @JoinColumn(name = "reward_id", unique = true)
    private Reward reward;    

}

File: EntreBicis_Frontend/app/src/main/java/cat/copernic/mbotana/entrebicis_frontend/reservation_management/domain/models/Reservation.kt
package cat.copernic.mbotana.entrebicis_frontend.reservation_management.domain.models

import cat.copernic.mbotana.entrebicis_frontend.reward_management.domain.models.Reward
import cat.copernic.mbotana.entrebicis_frontend.user_management.domain.models.User
import java.time.LocalTime

data class Reservation(
    var id: Long? = null,
    var reservationTime: LocalTime,
    var returnTime: LocalTime,
    var user: User,
    var reward: Reward
)
